clib/cMap.ml

```diff
@@ -164,6 +169,23 @@ struct
   | MEmpty -> 0
   | MNode (_, _, _, _, h) -> h
 
+  (* Filter based on a range *)
```

Comment:
This function looks like it is violating the invariants of the binary map structure, as it can generate unbalanced trees IIUC (can it?). Couldn't use the map API instead?

Reply:
Can it? Do subtrees of a tree violate some invariant? Note the use of `F.add`.

I'm afraid it cannot use the tree API and still be fast, TTBOMK.

Reply:
I don't see in you code how the call to filter_range is different to split, and furthermore the split function is going to be implemented by the OCaml stdlib in a similar way as filter_range... Anyways, if nobody complains much more I'll merge this PR soon.